dra-evolution/pkg/api/capacity_types.go

```diff
@@ -1,170 +1,205 @@
-	// is allocated.
-	ClaimUIDs []types.UID
+	// To be decided: We could split ResourcePool into spec and status or
+	// extend NamedDevice below so that it reflects the current state of
+	// that device. Like the spec that is data that is coming from a driver,
+	// so adding more information is not a conceptual change. Could be
+	// added in any future release as an extension.
+	//
+	// Much harder is tracking "allocated for". That would be useful to
+	// enable running multiple schedulers for different sets of nodes where
+	// all instances share ownership of some ResourcePool with network-attached
+	// devices. It also would enable a DRA driver's controller to co-exist
+	// with structured parameters in the scheduler.
+	//
+	// If we had transactions in the apiserver, we could combine a status
+	// update of the ResourcePool with a status update of the claim.  But
+	// we don't and adding it [is
+	// hard](https://kubernetes.slack.com/archives/C0EG7JC6T/p1714373064352099).
+	// Without transactions, there will be a risk of leaking resources
+	// and/or races around freeing leaked resources. This would be great to
+	// have in 1.31 because requiring schedulers to use the ResourcePool
+	// status for allocation will be a change of behavior, but it'll be
+	// hard to design and implement in time.
 }
 
-// Device is used to track individual devices in a pool.
-type Device struct {
-	// Name is a driver-specific identifier for the device.
-	// +required
-	Name string `json:"name"`
-
-	// Attributes contain additional metadata that can be used in
-	// constraints. If an attribute name overlaps with the pool attribute,
-	// the device attribute takes precedence.
+// ResourceModel must have one and only one field set.
+type ResourceModel struct {
+	// NamedDevices describes available devices by listing them.
 	//
 	// +optional
-	Attributes []Attribute `json:"attributes,omitempty"`
+	NamedDevices *NamedDevices `json:"namedDevices,omitempty" protobuf:"bytes,1,opt,name=namedResources"`
+}
 
-	// SharedResourcesConsumed contains the pooled resources that are
-	// consumed when this device is allocated.
+// NamedDevices is used in ResourceModel.
+type NamedDevices struct {
+	// CommonAttributes defines the set of attributes common to all devices in this ResourcePool.
+	// The name of each attribute must be unique.
 	//
+	// +listType=atomic
 	// +optional
-	SharedResourcesConsumed map[string]resource.Quantity `json:"sharedResourcesConsumed,omitempty"`
+	CommonAttributes []DeviceAttribute `json:"commonAttributes" protobuf:"bytes,2,name=commonAttributes"`
 
-	// ClaimResourcesProvided allows the definition of per-device resources
-	// that can be allocated in a manner similar to standard Kubernetes
-	// resources.
+	// A set of shared resources that get consumed as devices in this
+	// ResourcePool get allocated out.
+	SharedResources []DeviceResource `json:"sharedResources" protobuf:"bytes,3,name=sharedResources"`
```

Comment:
Let's ignore that for now because it is part of "consumable resources".

Reply:
@klueska: would it make sense to revert cd5ba42134bf730d34a1c5106968beef97353af4 and related commits to avoid getting sucked into a discussion of that more complex device model?

Reply:
Meaning we don't pursue support for partitionable devices in 1.31 at all?

Reply:
No. It just means that we stop debating that aspect until we have clarified the rest of the API.